keyboards/skyloong/gk87/bl_pro/bl_pro.c

```diff
@@ -0,0 +1,192 @@
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+       case MO(1):
+          if (record->event.pressed) {
+            FN_ON = 1;
+          } else {
+            FN_ON = 0;
+          }
+          return true;
+
+       case MO(2):
+           if (record->event.pressed) {
+             FN_ON = 1;
+           } else {
+             FN_ON = 0;
+           }
+          return true;
+
+       case MO(3):
+           if (record->event.pressed) {
+             FN_ON = 1;
+           } else {
+             FN_ON = 0;
+           }
+          return true;
+
+       case KC_LGUI:
+           if (FN_ON){
+              if ( record->event.pressed){
+                 WIN_LOCK = !WIN_LOCK ; //change win lock state
+                }
+              if (!WIN_LOCK) {
+                 IND = IND & (~WINLK_ON);   //Close win lock display
+                 s_serial_to_parallel(IND);
+                 gpio_write_pin_high(WIN_LOCK_PIN);
+                 return false; //windows key locked do nothing
+                }
+              s_serial_to_parallel(IND);
+            }
+           if (WIN_LOCK) {
+               IND = IND | WINLK_ON;   //Open win lock display
+               s_serial_to_parallel(IND);
+               gpio_write_pin_low(WIN_LOCK_PIN);
+               return false; //windows key locked do nothing
+            }
+           return true;  // continue all further processing of this key
+
+       case DF(0):
+           set_single_persistent_default_layer(0);
+           return true;
+
+       case DF(1):
+           set_single_persistent_default_layer(1);
+           return true;
+
+       case BL_TOGG:
+           if (record->event.pressed){
+               if(is_backlight_breathing()) {
+                    backlight_disable_breathing();
+                    DIS_BRETH = 1;
+
+                }else if(DIS_BRETH && !(is_backlight_enabled())){
+                    backlight_enable_breathing();
+                    DIS_BRETH = 0;
+                }
+           }
+         return true;
+
+       case BL_BRTG:
+           if (record->event.pressed){
+               if(DIS_BRETH || !(is_backlight_enabled())) {
+                 return false;
+                }
+           }
+         return true;
+
+       case BL_UP:
+           DIS_BRETH = 0;
+         return true;
+
+       case BL_DOWN:
+           if (record->event.pressed){
+               if(DIS_BRETH || !(is_backlight_enabled())) {
+                 return false;
+                }
+           }
+         return true;
+
+    default:
+      return true;
+    }
+   s_serial_to_parallel(IND);
+}
+
+void suspend_power_down_kb() {
+    SLEEP = 1;
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb() {
+    SLEEP = 0;
+    gpio_write_pin(LED_CAPS_LOCK_PIN, !host_keyboard_led_state().caps_lock);
+    gpio_write_pin(LED_SCROLL_LOCK_PIN, !host_keyboard_led_state().scroll_lock);
+    gpio_write_pin(WIN_LOCK_PIN, !WIN_LOCK);
```

Comment:
As above, using the keycodes, you could do this:
```suggestion
    gpio_write_pin(WIN_LOCK_PIN, !keycode_config.no_gui);
```